---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%",
  dpi = 400
)
```

# sisti

<!-- badges: start -->
[![CRAN status](https://www.r-pkg.org/badges/version/sisti)](https://CRAN.R-project.org/package=sisti)
[![R-CMD-check](https://github.com/ramiromagno/sisti/actions/workflows/R-CMD-check.yaml/badge.svg)](https://github.com/ramiromagno/sisti/actions/workflows/R-CMD-check.yaml)
<!-- badges: end -->

`{sisti}` provides real-time PCR data sets by Sisti et al. (2010) in tidy
format as one single table (also named) `sisti`.

The primary dataset comprises a ten-fold dilution series spanning copy numbers
from $3.14 \times 10^7$ to $3.14 \times 10^2$, with twelve replicates per
concentration. These samples are based on a pGEM-T Promega plasmid containing a
104 bp fragment of the mitochondrial gene NADH dehydrogenase 1 (MT-ND1),
amplified using the ND1/ND2 primer pair. The remaining three datasets contain
qPCR results in the presence of specific PCR inhibitors: tannic acid,
immunoglobulin G (IgG), and quercetin, respectively, to assess their effects on
the amplification process.

Each original data set can be obtained by filtering `sisti` by `plate` (see
below).

## Installation

``` r
install.packages("sisti")
```

## Data

Each sample group is defined by the inhibitor ("none", "tannic acid", "IgG" or
"quercitin"), respective inhibitor concentration, and initial amplicon copy
number.

```{r}
library(ggplot2)
library(dplyr, warn.conflicts = FALSE)
library(sisti)

sisti |>
  dplyr::distinct(plate, inhibitor, inhibitor_conc, copies, dilution) |>
  print(n = Inf)
```

Here is the number of replicates per group:

```{r}
sisti |>
  dplyr::distinct(plate, inhibitor, inhibitor_conc, copies, dilution, replicate) |>
  dplyr::count(plate, inhibitor, inhibitor_conc, copies, dilution) |>
  print(n = Inf)
```


### Standard dilution series

Most concentrated set of samples in the dilution series have $3.14 Ã— 10^7$
copies of the NADH dehydrogenase 1 (MT-ND1) amplicon. Following samples in
the series are ten-fold dilutions.

```{r}
sisti |>
  dplyr::filter(plate == "calibration") |>
  ggplot(aes(
    x = cycle,
    y = fluor,
    group = interaction(replicate, copies),
    col = as.factor(copies)
  )) +
  geom_line(linewidth = 0.1) +
  geom_point(size = 0.05) +
  labs(color = "Copy number")
```

### Inhibition by tannic acid

```{r}
sisti |>
  dplyr::filter(plate == "tannic acid") |>
  ggplot(aes(
    x = cycle,
    y = fluor,
    group = interaction(replicate, inhibitor_conc),
    col = as.factor(inhibitor_conc)
  )) +
  geom_line(linewidth = 0.1) +
  geom_point(size = 0.05) +
  labs(color = "Tannic acid conc (mg/mL)")
```

### Inhibition by immunoglobulin G (IgG)

```{r}
sisti |>
  dplyr::filter(plate == "IgG") |>
  ggplot(aes(
    x = cycle,
    y = fluor,
    group = interaction(replicate, inhibitor_conc),
    col = as.factor(inhibitor_conc)
  )) +
  geom_line(linewidth = 0.1) +
  geom_point(size = 0.05) +
  labs(color = "IgG conc (mg/mL)")
```

### Inhibition by quercitin

```{r}
sisti |>
  dplyr::filter(plate == "quercitin") |>
  ggplot(aes(
    x = cycle,
    y = fluor,
    group = interaction(replicate, inhibitor_conc),
    col = as.factor(inhibitor_conc)
  )) +
  geom_line(linewidth = 0.1) +
  geom_point(size = 0.05) +
  labs(color = "Quercitin conc (mg/mL)")
```

## Code of Conduct
  
Please note that the sisti project is released with a [Contributor Code of
Conduct](https://rmagno.eu/sisti/CODE_OF_CONDUCT.html). By contributing to this
project, you agree to abide by its terms.

## References

Davide Sisti, Michele Guescini, Marco BL Rocchi, Pasquale Tibollo, Mario D'Atri
and Vilberto Stocchi. _Shape based kinetic outlier detection in real-time PCR_.
BMC Bioinformatics 11:186 (2010). doi:
[10.1186/1471-2105-11-186](https://doi.org/10.1186/1471-2105-11-186).
